README.md

```diff
@@ -31,51 +31,94 @@
+
+### Further reading
 If you want more info about MAGE, check out the official [MAGE Documentation](https://docs.memgraph.com/mage/).
 
+### Algortihm proposition
 Furthermore, if you have an **algorithm proposition**, please fill in the survey on [**mage.memgraph.com**](https://mage.memgraph.com/).
 
 ## How to install?
 
-To build and install MAGE query modules you will need: **Python3**, **Make**, **CMake**, **Clang** and **Rust**. Also, you will need a runnable Memgraph instance, whether locally or via Docker image. [Download](https://memgraph.com/download), and [install](https://docs.memgraph.com/memgraph/getting-started/installation/) **Memgraph**.
+To build and install MAGE query modules you will need: **Python3**, **Make**, **CMake**, **Clang** and **Rust**. 
+Also, you will need a runnable Memgraph instance, whether locally or via Docker image. 
+There are few options to [download](https://memgraph.com/download), and [install](https://docs.memgraph.com/memgraph/getting-started/installation/) **Memgraph**
+depending on MAGE installation preferences.
+
+### 1. Installing MAGE with Docker image of Memgraph 
+For this part you will need Docker image of Memgraph.
+
+**Make sure to have `memgraph:latest` Docker image with following command:**
+
+```shell
+docker inspect memgraph
+```
+
 
-### Installing MAGE with Docker
-#### Docker Hub
+#### a) Install MAGE with Docker Hub
 
 **1.** This command downloads and runs Memgraph image with **MAGE** algorithms:
 ```
 docker run -p 7687:7687 memgraph/memgraph-mage
 ```
+This command will make use of your `memgraph:latest` image, download MAGE algorithms from Docker hub
+and add them to query modules directory, so they can be loaded with Memgraph.
+
+> Note: If you made any changes in directory on MAGE's algorithms, you should use b) option - Local Build of MAGE image
+#### b) Local Build of MAGE image from cloned repo
 
-#### Local Build
+Make sure that you have cloned MAGE Github repo and positioned yourself inside repo in terminal.
 
-**1.** Make sure to have `memgraph:latest` Docker image.
-**2.** Build **MAGE** tagged Docker image.
+**1.** Build **MAGE** Docker image with following command:
 ```
 docker build . -t memgraph-mage
 ```
 
-**3.** Start Memgraph with the following command and enjoy **MAGE**:
+**2.** Start Memgraph with the following command and enjoy **MAGE**:
 ```
 docker run -p 7687:7687 memgraph-mage
 ```
 
-### Installing MAGE locally
-**1.** Run the `build` script. It will generate a `dist` directory with all the needed files. If you want to skip step 2)
-you can add optional -p (--path) flag which represents where will contents of `dist` directory be copied.
+This will build any new algorithm added to MAGE, and load it inside Memgraph.
+
+### 2. Installing MAGE locally with Linux package of Memgraph
+
+In second option to run MAGE, you need Linux based Memgraph. Since Memgraph needs to load MAGE's modules, there is the
```

Comment:
Add link to this type of installation in MG documentation. This seems hard to follow for someone not understanding what "Linux based Memgraph" actually is